staging/src/k8s.io/apimachinery/pkg/runtime/conversion.go

```diff
@@ -82,32 +84,91 @@ func Convert_Slice_string_To_int(input *[]string, out *int, s conversion.Scope)
 	return nil
 }
 
+func string_to_bool(in string) bool {
+	switch strings.ToLower(in) {
+	case "false", "0":
+		return false
+	default:
+		return true
+	}
+}
+
+// Convert_string_To_bool will convert a string parameter to boolean.
+// Only a value of "false", or a value of "0" resolve to false.
+// Any other value (including empty string) resolves to true.
+func Convert_string_To_bool(in *string, out *bool, s conversion.Scope) error {
```

Comment:
does this match what reflective conversion was doing (especially the zero-length string aspect)?

Reply:
This didn't work before (reflective conversion is not converting string to bool).

Regarding zero-length - I'm making this consistent with conversion of string slice.

Reply:
> This didn't work before (reflective conversion is not converting string to bool).

If reflective conversion didn't do this, and this isn't being called in generated code in this PR, why is it being added?

> Regarding zero-length - I'm making this consistent with conversion of string slice.

That's tricky. A string slice of `[""]` is not the zero-value of the slice, and when read from URL parameters, means that the parameter was present in the URL with no value. A string of "" is the zero value, and would intuitively convert to `false`. I think we need more information about how this is being called/used to know what the correct conversion here is

Reply:
Removing (given that I was adding similar for int64 I though it may be useful, but apparently it's more controversial). So given it's not really needed, I'm removing it.